### PR TITLE
scripts: support soc/boards in Zephyr modules

### PR DESCRIPTION
**Pull request by tejlmand**
Extend check_compliance with support for soc and boards defined Zephyr modules.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

File: scripts/ci/check_compliance.py [hunk 1]
```diff
@@ -363,7 +363,7 @@ def get_modules(self, modules_file, settings_file):
                 ))
             fp_module_file.write(content)
 
-    def get_kconfig_dts(self, kconfig_dts_file, settings_file):
+    def get_module_setting_root(self, root, settings_file):
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply scripts/ci/check_compliance.py:366]: fixed, thanks for noticing.

File: scripts/ci/check_compliance.py [hunk 2]
```diff
@@ -363,6 +363,27 @@ def get_modules(self, modules_file, settings_file):
                 ))
             fp_module_file.write(content)
 
+    def get_module_setting_root(self, root, settings_file):
+        """
+        Parse the Zephyr module generated settings file given by 'settings_file'
+	and return all root settings defined by 'root'.
```
Comment (keith-zephyr) [inline scripts/ci/check_compliance.py:369]: Nit - incorrect indentation.

Comment (tejlmand) [reply scripts/ci/check_compliance.py]: Fixed.

**Comment by MaureenHelm** [comment]
@keith-zephyr please revisit

**Review by keith-zephyr — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**